handler/proxy.go

```diff
@@ -209,6 +222,12 @@ func (p *Proxy) roundtrip(rw http.ResponseWriter, req *http.Request) {
 		outreq.Header.Set("X-Forwarded-For", clientIP)
 	}
 
+	if ReClientSupportsGZ.MatchString(req.Header.Get(AcceptEncodingHeader)) {
```

Comment:
This code should be within `Director` so it's clear and at a defined place for modifying the outgoing request.

In this case, the client request gets modified and not the outgoing `outreq` which gets setup by the `Director` method. That means that the clients AE header gets passed to the origin. So an additional test should cover if a client sends AE gzip, that we set it too or not.